[ROSBE] 2.2.0 Updates at it's finest

dreimer1986 opened the pull request:
Update changelog last time
Removed old hack
Readded dlls needed for cmake
Add more rights to Users in Install root dir.
Semi fix GIT mess

dreimer1986 committed:
[ROSBE]
Removed old hack
Readded dlls needed for cmake

dreimer1986 committed:
[ROSBE]
Removed old hack
Readded dlls needed for cmake

dreimer1986 committed:
[ROSBE]
Removed old hack
Readded dlls needed for cmake

dreimer1986 committed:
[ROSBE]
Removed old hack
Readded dlls needed for cmake

dreimer1986 committed:
[ROSBE]
Update changelog last time

dreimer1986 committed:
[ROSBE]
Update changelog last time

dreimer1986 committed:
[ROSBE]
Update changelog last time

dreimer1986 committed:
[ROSBE]
Update changelog last time

dreimer1986 committed:
[ROSBE]
This is what happens when you try to make everyone happy at once. Add more rights to Users in Install root dir.

dreimer1986 committed:
[ROSBE]
This is what happens when you try to make everyone happy at once. Add more rights to Users in Install root dir.

dreimer1986 committed:
[ROSBE]
This is what happens when you try to make everyone happy at once. Add more rights to Users in Install root dir.

dreimer1986 committed:
[ROSBE]
This is what happens when you try to make everyone happy at once. Add more rights to Users in Install root dir.

dreimer1986 committed:
[ROSBE]
Semi fix GIT mess

dreimer1986 committed:
[ROSBE]
Semi fix GIT mess

dreimer1986 committed:
[ROSBE]
Semi fix GIT mess

learn-more commented:
Your first commit (e9bf78b) is still with your nickname,
and by only having the component on the first line, the git history doesnt really read nicely:

![](https://i.imgur.com/r4AKEXa.png)

learn-more commented:
@dreimer1986 Any updates?

binarymaster commented on RosBE-Windows/RosBE.nsi:
```diff
@@ -169,6 +169,7 @@ Section -BaseFiles SEC01
     File /r Components\Bin\gdb.exe
     File /r Components\Bin\gdbserver.exe
     File /r Components\Bin\getdate.exe
+	File /r Components\Bin\libeay32.dll
```
Indentation problem here.

dreimer1986 committed:
[ROSBE] New Version preparation

Add AMD64 compiler to base setup

dreimer1986 committed:
[ROSBE] New Version preparation

Add AMD64 compiler to base setup

dreimer1986 committed:
[ROSBE] New Version preparation

Add AMD64 compiler to base setup

dreimer1986 committed:
[ROSBE] New Version preparation

Add AMD64 compiler to base setup

learn-more commented on RosBE-Windows/RosBE-Windows_RosBE.nsi:
```diff
@@ -0,0 +1,526 @@
+    SetAutoClose true
+SectionEnd
+
+;; HACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACKHACK
```
This might be a hack?

dreimer1986 replied:
This stuff is outdated and will be replaced soon. I still hate GIT btw!

dreimer1986 committed:
[ROSBE] merge mess!

dreimer1986 committed:
[ROSBE] merge mess!

dreimer1986 committed:
[ROSBE] merge mess!

dreimer1986 committed:
[ROSBE] merge mess!

dreimer1986 committed:
[ROSBE] More preparation for new version - Include AMD64 compiler into base package as optional component - Add new XP GlobalSign Certs

dreimer1986 committed:
[ROSBE] More preparation for new version - Include AMD64 compiler into base package as optional component - Add new XP GlobalSign Certs

dreimer1986 committed:
[ROSBE] More preparation for new version - Include AMD64 compiler into base package as optional component - Add new XP GlobalSign Certs

dreimer1986 committed:
[ROSBE] More preparation for new version - Include AMD64 compiler into base package as optional component - Add new XP GlobalSign Certs

dreimer1986 committed:
[ROSBE] More preparation for new version - Include AMD64 compiler into base package as optional component - Add new XP GlobalSign Certs

dreimer1986 committed:
[ROSBE] More preparation for new version - Include AMD64 compiler into base package as optional component - Add new XP GlobalSign Certs

dreimer1986 committed:
[ROSBE] More preparation for new version - Include AMD64 compiler into base package as optional component - Add new XP GlobalSign Certs

dreimer1986 committed:
[ROSBE] More preparation for new version - Include AMD64 compiler into base package as optional component - Add new XP GlobalSign Certs

dreimer1986 committed:
[ROSBE] forgot to remove old cert...

dreimer1986 committed:
[ROSBE] forgot to remove old cert...

dreimer1986 committed:
[ROSBE] forgot to remove old cert...

HBelusca commented on RosBE-Windows/RosBE.nsi:
```diff
@@ -183,8 +184,10 @@ Section -BaseFiles SEC01
     File /r Components\Bin\MSVCM90.dll
     File /r Components\Bin\msvcp60.dll
     File /r Components\Bin\MSVCP90.dll
+	File /r Components\Bin\MSVCP100.dll
```
Ditto

HBelusca commented on RosBE-Windows/RosBE.nsi:
```diff
@@ -183,8 +184,10 @@ Section -BaseFiles SEC01
+	File /r Components\Bin\MSVCP100.dll
     File /r Components\Bin\MSVCP120.dll
     File /r Components\Bin\MSVCR90.dll
+	File /r Components\Bin\MSVCR100.dll
```
Ditto

HBelusca commented on RosBE-Windows/RosBE.nsi:
```diff
@@ -197,8 +200,10 @@ Section -BaseFiles SEC01
     File /r Components\Bin\rquote.exe
     File /r Components\Bin\scut.exe
     File /r Components\Bin\sdiff.exe
+	File /r Components\Bin\ssleay32.dll
```
Ditto

HBelusca commented on RosBE-Windows/RosBE.nsi:
```diff
@@ -197,8 +200,10 @@ Section -BaseFiles SEC01
+	File /r Components\Bin\ssleay32.dll
     File /r Components\Bin\tee.exe
     File /r Components\Bin\wget.exe
+	File /r Components\Bin\zlib1.dll
```
Ditto

HBelusca commented on RosBE-Windows/RosBE.nsi:
```diff
@@ -275,20 +308,25 @@ Section /o "PowerShell Version" SEC05
                 CreateShortCut "$SMPROGRAMS\$ICONS_GROUP\ReactOS Build Environment ${PRODUCT_VERSION}.lnk" "$SYSDIR\cmd.exe" '/t:0A /k "$INSTDIR\RosBE.cmd"' "$INSTDIR\rosbe.ico"
             IfFileExists "$INSTDIR\RosBE.ps1" 0 +2
                 CreateShortCut "$SMPROGRAMS\$ICONS_GROUP\ReactOS Build Environment ${PRODUCT_VERSION} - PS.lnk" "$SYSDIR\WindowsPowerShell\v1.0\powershell.exe" "-noexit &'$INSTDIR\RosBE.ps1'" "$INSTDIR\rosbe.ico"
+			IfFileExists "$INSTDIR\amd64\*" 0 +6
```
Are there indentation problems in the following lines?

HBelusca commented on RosBE-Windows/RosBE.nsi:
```diff
@@ -299,35 +337,45 @@ Section -StartMenuShortcuts SEC06
             CreateShortCut "$DESKTOP\ReactOS Build Environment ${PRODUCT_VERSION}.lnk" "$SYSDIR\cmd.exe" '/t:0A /k "$INSTDIR\RosBE.cmd"' "$INSTDIR\rosbe.ico"
         IfFileExists "$INSTDIR\RosBE.ps1" 0 +2
             CreateShortCut "$DESKTOP\ReactOS Build Environment ${PRODUCT_VERSION} - PS.lnk" "$SYSDIR\WindowsPowerShell\v1.0\powershell.exe" "-noexit &'$INSTDIR\RosBE.ps1'" "$INSTDIR\rosbe.ico"
+		IfFileExists "$INSTDIR\amd64\*" 0 +6
```
Are there indentation problems in the following lines?

HBelusca commented on RosBE-Windows/RosBE.nsi:
```diff
@@ -358,6 +406,10 @@ Function un.onInit
         Delete /REBOOTOK "$QUICKLAUNCH\ReactOS Build Environment ${PRODUCT_VERSION}.lnk"
         Delete /REBOOTOK "$DESKTOP\ReactOS Build Environment ${PRODUCT_VERSION} - PS.lnk"
         Delete /REBOOTOK "$QUICKLAUNCH\ReactOS Build Environment ${PRODUCT_VERSION} - PS.lnk"
+		Delete /REBOOTOK "$DESKTOP\ReactOS Build Environment ${PRODUCT_VERSION} AMD64.lnk"
```
Indentation problem.

HBelusca commented on RosBE-Windows/RosBE.nsi:
```diff
@@ -299,35 +337,45 @@ Section -StartMenuShortcuts SEC06
             CreateShortCut "$QUICKLAUNCH\ReactOS Build Environment ${PRODUCT_VERSION}.lnk" "$SYSDIR\cmd.exe" '/t:0A /k "$INSTDIR\RosBE.cmd"' "$INSTDIR\rosbe.ico"
         IfFileExists "$INSTDIR\RosBE.ps1" 0 +2
             CreateShortCut "$QUICKLAUNCH\ReactOS Build Environment ${PRODUCT_VERSION} - PS.lnk" "$SYSDIR\WindowsPowerShell\v1.0\powershell.exe" "-noexit &'$INSTDIR\RosBE.ps1'" "$INSTDIR\rosbe.ico"
+		IfFileExists "$INSTDIR\amd64\*" 0 +6
```
Are there indentation problems in the following lines?

dreimer1986 committed:
[ROSBE] Fix indentation...

dreimer1986 committed:
[ROSBE] Fix indentation...

dreimer1986 committed:
[ROSBE] Fix indentation...

dreimer1986 committed:
[ROSBE] Fix indentation...

dreimer1986 committed:
[ROSBE] Fix removal of desktop shortcuts on uninstall. No good idea to remove from current user while the shortcuts are on public desktop... Fix removal of a few shortcuts even if you say NO on uninstall.

dreimer1986 committed:
[ROSBE] Fix removal of desktop shortcuts on uninstall. No good idea to remove from current user while the shortcuts are on public desktop... Fix removal of a few shortcuts even if you say NO on uninstall.

dreimer1986 committed:
[ROSBE] Fix removal of desktop shortcuts on uninstall. No good idea to remove from current user while the shortcuts are on public desktop... Fix removal of a few shortcuts even if you say NO on uninstall.

dreimer1986 committed:
[ROSBE] Fix removal of desktop shortcuts on uninstall. No good idea to remove from current user while the shortcuts are on public desktop... Fix removal of a few shortcuts even if you say NO on uninstall.

HBelusca commented:
Does this new version of RosBE work on ReactOS?

dreimer1986 committed:
(RosBE) First updates for the other binaries. First parts of changelog.

dreimer1986 committed:
(RosBE) First updates for the other binaries. First parts of changelog.

dreimer1986 committed:
(RosBE) First updates for the other binaries. First parts of changelog.

dreimer1986 committed:
(RosBE) First updates for the other binaries. First parts of changelog.

dreimer1986 committed:
(RosBE) First updates for the other binaries. First parts of changelog.

dreimer1986 committed:
(RosBE) First updates for the other binaries. First parts of changelog.

dreimer1986 committed:
(RosBE) First updates for the other binaries. First parts of changelog.

dreimer1986 committed:
(RosBE) First updates for the other binaries. First parts of changelog.

dreimer1986 committed:
[ROSBE] Hello new dependencies. Wanna bet I forgot one? ^^

dreimer1986 committed:
[ROSBE] Hello new dependencies. Wanna bet I forgot one? ^^

dreimer1986 committed:
[ROSBE] Hello new dependencies. Wanna bet I forgot one? ^^

dreimer1986 committed:
[ROSBE] Hello new dependencies. Wanna bet I forgot one? ^^

dreimer1986 commented:
![VirtualBox_ROS_28_04_2019_01_26_43](https://user-images.githubusercontent.com/29841368/56856157-b963b700-6954-11e9-9a3d-94997d38ea9b.png)

As long as you ignore wget causing trouble in ws2_32 thx to being made for anyting but Windows XP.... Looks good for now. And if you wanna know why: https://eternallybored.org/misc/wget/ XP support dropped...

dreimer1986 committed:
[ROSBE] Fix Bison path, update changelog + remove wget update for XP and ROS support, try to persuade setup to create a lower case bin folder to make new cmake happy, fix all errors showing up in PS version due to bad regex and not enough file access rights...

dreimer1986 committed:
[ROSBE] Fix Bison path, update changelog + remove wget update for XP and ROS support, try to persuade setup to create a lower case bin folder to make new cmake happy, fix all errors showing up in PS version due to bad regex and not enough file access rights...

dreimer1986 committed:
[ROSBE] Fix Bison path, update changelog + remove wget update for XP and ROS support, try to persuade setup to create a lower case bin folder to make new cmake happy, fix all errors showing up in PS version due to bad regex and not enough file access rights...

dreimer1986 committed:
[ROSBE] Fix Bison path, update changelog + remove wget update for XP and ROS support, try to persuade setup to create a lower case bin folder to make new cmake happy, fix all errors showing up in PS version due to bad regex and not enough file access rights...

dreimer1986 committed:
[ROSBE] Downgrade CMake to maintain XP/ROS support... and fix a typo

dreimer1986 committed:
[ROSBE] Downgrade CMake to maintain XP/ROS support... and fix a typo

dreimer1986 committed:
[ROSBE] Downgrade CMake to maintain XP/ROS support... and fix a typo

dreimer1986 committed:
[ROSBE] Downgrade CMake to maintain XP/ROS support... and fix a typo

dreimer1986 committed:
[ROSBE] Check through all needed CRTs and remove the no more needed ones. Dedicated to Thomas Faber ^^

dreimer1986 committed:
[ROSBE] Check through all needed CRTs and remove the no more needed ones. Dedicated to Thomas Faber ^^

dreimer1986 committed:
[ROSBE] Check through all needed CRTs and remove the no more needed ones. Dedicated to Thomas Faber ^^

dreimer1986 committed:
[ROSBE] Check through all needed CRTs and remove the no more needed ones. Dedicated to Thomas Faber ^^

dreimer1986 committed:
[ROSBE]
Update changelog once more

dreimer1986 committed:
[ROSBE]
Update changelog once more

dreimer1986 committed:
[ROSBE]
Update changelog once more

dreimer1986 committed:
[ROSBE]
Update changelog once more

dreimer1986 committed:
[ROSBE] Add new dependencies for CMAKE build, update changelog and Readme

dreimer1986 committed:
[ROSBE] Add new dependencies for CMAKE build, update changelog and Readme

dreimer1986 committed:
[ROSBE] Add new dependencies for CMAKE build, update changelog and Readme

dreimer1986 committed:
[ROSBE] Add new dependencies for CMAKE build, update changelog and Readme

dreimer1986 committed:
[ROSBE] Test commit to make GIT fuck up again... Change default install dir back to C:\RosBE

dreimer1986 committed:
[ROSBE] Test commit to make GIT fuck up again... Change default install dir back to C:\RosBE

dreimer1986 committed:
[ROSBE] Test commit to make GIT fuck up again... Change default install dir back to C:\RosBE

dreimer1986 committed:
[ROSBE] Test commit to make GIT fuck up again... Change default install dir back to C:\RosBE

ColinFinck committed:
Prepare the scripts for the RosBE 2.2 release.

ColinFinck committed:
Update README files for the RosBE-Unix 2.2 release.

ColinFinck committed:
[ROSBE-UNIX] Add Bison 3.5, Flex 2.6.4+ (latest Git), and update to CMake 3.16.2.

Latest Flex 2.6.4 release version doesn't build under Ubuntu 18.04, so I had to take the cutting-edge version.
Unrelated of this change, we also don't have a Bison and Flex build dependency anymore.

New packages are available at https://svn.reactos.org/RosBE-Sources/rosbe_2.2

ColinFinck committed:
[ROSBE-UNIX] Simplify CFLAGS handling and also add CXXFLAGS (for at least CMake).

ColinFinck committed:
[ROSBE-UNIX] Build CMake explicitly without SSL support.

This is required since https://gitlab.kitware.com/cmake/cmake/commit/dd3e4767863dfe1307d4dccd2445e5312b55473e
We are not using CMake to download anything anyway.

ColinFinck committed:
[ROSBE-UNIX] Don't export the CC and CXX variables, this clashes with the mingw-w64 build inside the gcc build.

ColinFinck committed:
[ROSBE-UNIX] Add the now required "m4" dependency for building Bison.

ColinFinck committed:
[ROSBE] Add a patch to fix the Bison 3.5 Win32 build the easy way.

Not suitable for upstream inclusion, but will be submitted as an inspiration to the winflexbison project at least.
New bison package is up at https://svn.reactos.org/RosBE-Sources/rosbe_2.2

ColinFinck committed:
[ROSBE-UNIX] Update README and add .gitignore

ColinFinck committed:
[ROSBE-WINDOWS] Sync Buildtoolchain with latest RosBE-Unix changes to provide Bison and Flex builds and an updated CMake.

Bison 3.5 doesn't build under Mingw-w64 out of the box, but the small patch "bison-3.5-reactos-fix-win32-build.patch" fixes that.
It then only needs the BISON_PKGDATADIR and M4 environment variables set correctly (due to poor relocatability of the built binary).

Flex 2.6.4 depends on several POSIX-isms, and the version from winflexbison is heavily modified, so I decided for an MSYS2 build.
This requires a different environment, which is why we have an additional "buildtoolchain-msys.sh" script now.
Let's hope they import some winflexbison changes in the future.

Updated builds are available at https://svn.reactos.org/RosBE-Temp

ColinFinck committed:
[ROSBE-UNIX] Update sanity checks in makepackage.sh

ColinFinck commented on RosBE-Windows/Powershell/RosBE.ps1:
```diff
@@ -69,7 +69,7 @@ $global:_BUILDBOT_SVNSKIPMAINTRUNK = "0"
 
 # Fix Bison package path (just in case RosBE is installed in a path which contains spaces)
-$ENV:BISON_PKGDATADIR = ((New-Object -ComObject Scripting.FileSystemObject).GetFolder("$_ROSBE_HOST_MINGWPATH\share\bison")).ShortPath
+$ENV:BISON_PKGDATADIR = ((New-Object -ComObject Scripting.FileSystemObject).GetFolder("$_ROSBE_BASEDIR\bin\data")).ShortPath
```
With my Bison 3.5 binary from https://svn.reactos.org/RosBE-Temp/RosBE-2.2_Dec2019.7z, this environment variable is required and the data path is share\bison.
We also need another environment variable "M4" with the full path to the m4.exe binary (which is also shipped in a defined version in RosBE-2.2_Dec2019.7z).

dreimer1986 replied:
Fixed I hope.

ColinFinck commented on RosBE-Windows/Root/ChangeLog.txt:
```diff
@@ -1,5 +1,23 @@
+  - i686-w64-mingw32 AND x86_64-w64_mingw32 (optional) in one setup!
+- Updated binutils to 2.32 (Colin Finck, Daniel Reimer)
+- Updated mingw-w64 to 6.0.0 (Colin Finck, Daniel Reimer)
+- Updated Flex to 2.6.4 (Daniel Reimer) (https://sourceforge.net/projects/winflexbison/)
```
It's Bison 3.5 and Flex 2.6.4+ (revision 8b1fbf674f2e038df9cf1fe7725617e3837ae2a9) with my updated binaries from https://svn.reactos.org/RosBE-Temp/RosBE-2.2_Dec2019.7z
We are using the very same versions for RosBE-Unix now and can therefore safely make Bison and Flex a build dependency.

The winflexbison sources are heavily modified to an extent, where they don't build on anything else than Windows/MSVC anymore and many code paths needed to be disabled for this.
Therefore, winflexbison wasn't suitable for sourcing reliable versions that can be shared between RosBE-Windows and RosBE-Unix

dreimer1986 replied:
Fixed I hope.

ColinFinck commented on RosBE-Windows/Root/RosBE.cmd:
```diff
@@ -90,7 +90,7 @@ set _BUILDBOT_SVNSKIPMAINTRUNK=0
 
 :: Fix Bison package path (just in case RosBE is installed in a path which contains spaces)
-set BISON_PKGDATADIR=%~ds0%~sp0%i386\share\bison
+set BISON_PKGDATADIR=%~ds0%~sp0%bin\data
```
Requires same change as above

dreimer1986 replied:
Fixed I hope.

GeoB99 committed:
[ROSBE-UNIX] Accept Yes/No answers with upper case letters (#6)

So far, the script can only check if the user answered yes or no, without capital letters. That would make the script exit unexpectedly if the user, for instance, answered Yes or YES in the input.

dreimer1986 committed:
Updates for Dec 2019 bison/flex/m4 inclusion, untested and setup still needs updating!
(GIT still sucks!)

dreimer1986 committed:
Updates for Dec 2019 bison/flex/m4 inclusion, untested and setup still needs updating!
(GIT still sucks!)

dreimer1986 committed:
Updates for Dec 2019 bison/flex/m4 inclusion, untested and setup still needs updating!
(GIT still sucks!)

dreimer1986 committed:
Updates for Dec 2019 bison/flex/m4 inclusion, untested and setup still needs updating!
(GIT still sucks!)

HBelusca commented:
Do the RosBE 2.2.0 components work on ReactOS?

dreimer1986 committed:
Update RosBE NSIS Script to include a host tools lib and include as needed by new binaries, remove old bison share not needed anymore, add new binaries

dreimer1986 committed:
Update RosBE NSIS Script to include a host tools lib and include as needed by new binaries, remove old bison share not needed anymore, add new binaries

dreimer1986 committed:
Update RosBE.nsi

-- New shortcut I will use from now on: GIT POS (Pile of SHIT) --
GIT POS caused crap again, thus I fix it by copying over RosBE.nsi

dreimer1986 committed:
Update RosBE.nsi

-- New shortcut I will use from now on: GIT POS (Pile of SHIT) --
GIT POS caused crap again, thus I fix it by copying over RosBE.nsi

dreimer1986 committed:
Update RosBE.nsi

GIT POS fix #2

dreimer1986 committed:
Update RosBE.nsi

GIT POS fix #2

dreimer1986 committed:
Merge branch 'master' of https://github.com/dreimer1986/RosBE

dreimer1986 committed:
Update README.*

dreimer1986 committed:
Update README.*

dreimer1986 committed:
Fix for NSIS 3.05 by replacing the dated macro for PATH var manipulation with the very recent plugin "EnvVar".
Make sure that the certificate updater does NOTHING on anything newer than XP.

dreimer1986 committed:
Fix for NSIS 3.05 by replacing the dated macro for PATH var manipulation with the very recent plugin "EnvVar".
Make sure that the certificate updater does NOTHING on anything newer than XP.

dreimer1986 committed:
- M4 path needs the FULL path, not the folder the exe is inside...
- Fix a 'maybe' problem with multi arch and options.exe
- Remove color codes from shortcuts. We do things the same way now on PS and Batch

dreimer1986 committed:
- M4 path needs the FULL path, not the folder the exe is inside...
- Fix a 'maybe' problem with multi arch and options.exe
- Remove color codes from shortcuts. We do things the same way now on PS and Batch

dreimer1986 commented:
https://dreimer.eu/rosbe/RosBE-2.2.0.exe Test it!

Zero3K commented:
Its stating that the CMake at /RosBE/bin isn't the custom one even though it is (since it has a date of 12/26/2019).

dreimer1986 commented:
> Its stating that the CMake at /RosBE/bin isn't the custom one even though it is (since it has a date of 12/26/2019).

Known. This check will be removed

ColinFinck committed:
[ROSBE-WINDOWS] Add precise checks for the correct MSYS environment.

Extravert-ir committed:
[ROSBE-WINDOWS] Update dependencies versions

Extravert-ir committed:
[ROSBE] Build GCC with --enable-sjlj-exceptions.
In fact it was built like that before as well
(probably because GCC 4.7.2 don't support dwarf2 exceptions)
Binaries built with RosBE and dwarf2 exceptions don't work for some unknown (yet) reason

dreimer1986 committed:
Merge branch 'master' of https://github.com/dreimer1986/RosBE

dreimer1986 committed:
[CHANGELOG]
Another Update of used versions

dreimer1986 committed:
Bye Cert Installer for XP, not needed anymore
Cleaned up some dependencies noone needs anymore.
Fixed some line endings

dreimer1986 committed:
[ROSBE] Finally, release is close

dreimer1986 committed:
This is the end ^^
